src/PhpDoc/ResolvedPhpDocBlock.php

```diff
@@ -380,4 +384,178 @@ public function getTemplateTypeMap(): TemplateTypeMap
 		return $this->templateTypeMap;
 	}
 
+	/**
+	 * @param array<int, self> $parents
+	 * @param array<int, PhpDocBlock> $parentPhpDocBlocks
+	 * @return self
+	 */
+	public function cloneAndMerge(array $parents, array $parentPhpDocBlocks): self
+	{
+		$result = clone $this;
+		$result->mergeTags($parents, $parentPhpDocBlocks);
+		return $result;
+	}
+
+	/**
+	 * @param array<int, self> $parents
+	 * @param array<int, PhpDocBlock> $parentPhpDocBlocks
+	 */
+	private function mergeTags(array $parents, array $parentPhpDocBlocks): void // phpcs:disable
```

Comment:
I haven't seed phpcs:disable in your code but not detecting clones is a known issue with phpcs:
https://github.com/slevomat/coding-standard/issues/642

Reply:
I don't want to use `clone` anyway. I always want to call constructor explicitly. Thanks.